xdsl/dialects/tensor.py

```diff
@@ -293,6 +298,97 @@ def verify_(self) -> None:
             )
 
 
+@irdl_op_definition
+class ExpandShapeOp(IRDLOperation):
+    """
+    Operation to produce a tensor with a higher rank
+    """
+
+    # Constant value used to denote dynamic indices in offsets, sizes, and strides.
+    # Same constant as in MLIR.
+    DYNAMIC_INDEX: ClassVar[int] = -9223372036854775808
+
+    name = "tensor.expand_shape"
+
+    src = operand_def(TensorType)
+    dynamic_output_shape = var_operand_def(IndexType)  # Dynamic dims
+
+    reassociation = prop_def(ReassociationAttr)
+
+    output_shape = prop_def(DenseArrayBase)
+
+    result = result_def(AnyAttr())
+
+    def __init__(
+        self,
+        src: SSAValue | Operation,
+        dynamic_output_shape: Sequence[SSAValue],
+        reassociation: ReassociationAttr,
+        static_output_shape: Sequence[int] | DenseArrayBase,
+        result_type: TensorType[Attribute],
+    ):
+        if not isinstance(static_output_shape, DenseArrayBase):
+            static_output_shape = DenseArrayBase.create_dense_int(
+                i64, static_output_shape
+            )
+
+        super().__init__(
+            operands=[src, dynamic_output_shape],
+            result_types=[result_type],
+            properties={
+                "reassociation": reassociation,
+                "output_shape": static_output_shape,
+            },
+        )
+
+    @classmethod
+    def parse(cls, parser: Parser) -> Self:
+        src_operand = parser.parse_unresolved_operand()
+
+        reassociation = parser.parse_attribute()
+        parser.parse_characters("output_shape")
+        index = IndexType()
+
+        # Parse shape: mixture of ints and SSA values
+        dyn_shape, static_shape = parse_dynamic_index_list_without_types(
+            parser, dynamic_index=cls.DYNAMIC_INDEX
+        )
+
+        dyn_shape = parser.resolve_operands(
+            dyn_shape, (index,) * len(dyn_shape), parser.pos
+        )
+
+        parser.parse_punctuation(":")
+        src_type = parser.parse_type()
+        parser.parse_characters("into")
+        result_type = parser.parse_type()
+        src = parser.resolve_operand(src_operand, src_type)
+
+        shape_attr = DenseArrayBase.create_dense_int(i64, static_shape)
+
+        reassociation = cast(ReassociationAttr, reassociation)
+        result_type = cast(TensorType[Attribute], result_type)
+
+        return cls(src, dyn_shape, reassociation, shape_attr, result_type)
+
+    def print(self, printer: Printer):
+        printer.print(" ")
```

Comment:
We're currently removing uses of this function, please use print_string instead
```suggestion
        printer.print_string(" ")
```